tests/integration/test_load_and_serve_agent.py

```diff
@@ -0,0 +1,185 @@
+import logging
+import os
+import subprocess
+from datetime import datetime
+from pathlib import Path
+from typing import Any
+
+import pytest
+from litellm.utils import validate_environment
+
+import any_agent.serving
+from any_agent import AgentConfig, AgentFramework, AnyAgent, TracingConfig
+from any_agent.config import MCPStdio, ServingConfig
+
+from opentelemetry.sdk.trace.export.in_memory_span_exporter import (
+    InMemorySpanExporter,
+)
+from opentelemetry.sdk.trace import ReadableSpan
+
+if any_agent.serving.serving_available:
+    from common.client import A2AClient
+    from common.types import Message, TextPart
+
+FORMAT = "%(message)s"
+logging.basicConfig(
+    format=FORMAT,
+    datefmt="[%X]",
+)
+logger = logging.getLogger("any_agent_test")
+logger.setLevel(logging.DEBUG)
+
+"""
+def organize(items: list[AgentSpan]) -> None:
+    traces = {}
+    for trace in items:
+        k = trace.context.span_id
+        trace.attributes[CHILD_TAG] = {}
+        traces[k] = trace
+    for trace in items:
+        if trace.parent:
+            parent_k = trace.parent.span_id
+            if parent_k:
+                traces[parent_k].attributes[CHILD_TAG][trace.context.span_id] = trace
+            else:
+                traces[None] = trace
+"""
+class SpanNode:
+    def __init__(span):
+        _span: ReadableSpan = span
+        _children: list["SpanNode"] = []
+    def add_span(child_span):
+        """Add child span sorted according to timestamps"""
+        ...
+
```

Comment:
Some comments to explain what all this is would be useful

Reply:
It's still in progress :) But I was waiting for your feedback on the work so far to see if I should go through this route.